### PR TITLE
update dry-run metric prefix and also add e2e tests

### PR DESCRIPTION
**Pull request by yangminzhu**
This PR updates to use the correct metric prefix for dry-run results and also add e2e tests with stackdriver.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: tests/integration/telemetry/stackdriver/stackdriver_filter_dry_run_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+)
+
+const (
+	dryRunServerLogEntry         = "testdata/security_authz_dry_run/server_access_log.json.tmpl"
+	dryRunTCPServerLogEntry      = "testdata/security_authz_dry_run/tcp_server_access_log.json.tmpl"
+	dryRunAuthorizationPolicy    = "testdata/security_authz_dry_run/authorization_policy.yaml.tmpl"
+	dryRunTCPAuthorizationPolicy = "testdata/security_authz_dry_run/tcp_authorization_policy.yaml.tmpl"
+)
+
+// TestStackdriverAuthzDryRun verifies that stackdriver WASM filter exports dry-run metrics with expected labels.
```
Comment (bianpengyuan) [inline tests/integration/telemetry/stackdriver/stackdriver_filter_dry_run_test.go:42]:
```suggestion
// TestStackdriverAuthzDryRun verifies that stackdriver WASM filter exports dry-run logss with expected labels.
```

File: tests/integration/telemetry/stackdriver/stackdriver_filter_dry_run_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+					return nil
+				})
+			}
+			if err := g.Wait(); err != nil {
+				t.Fatalf("test failed: %v", err)
+			}
+		})
+}
+
+// TestTCPStackdriverAuthzDryRun verifies that stackdriver TCP filter works.
```
Comment (bianpengyuan) [inline tests/integration/telemetry/stackdriver/stackdriver_filter_dry_run_test.go:70]: Please update the comment here.

File: tests/integration/telemetry/stackdriver/testdata/security_authz_dry_run/tcp_server_access_log.json.tmpl [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        "source_canonical_revision": "v1",
+        "source_canonical_service": "clt-{{ .ClusterName }}",
+        "source_namespace": "{{ .EchoNamespace }}",
+        "source_principal": "spiffe://{{ .TrustDomain }}/ns/{{ .EchoNamespace }}/sa/default",
+        "source_version": "v1",
+        "source_workload": "clt-{{ .ClusterName }}-v1",
+        "protocol": "tcp",
+        "log_sampled": "false",
+        "requested_server_name": "outbound_.9090_._.srv.{{ .EchoNamespace }}.svc.cluster.local",
+        "upstream_cluster": "inbound|9000||"
```
Comment (bianpengyuan) [inline tests/integration/telemetry/stackdriver/testdata/security_authz_dry_run/tcp_server_access_log.json.tmpl:27]: missed a comma here.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**